sql: deflake TestSchemaChangeRetry

ajwerner opened the pull request:
Now that GC happens eagerly in the GC job, we need to hold off the GC job to
make sure that key counts match the test's expectations.

Fixes flakes like [this one](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/6093943)

Release justification: Testing only change. 

Release note: None

ajwerner committed:
sql: deflake TestSchemaChangeRetry

Now that GC happens eagerly in the GC job, we need to hold off the GC job to
make sure that key counts match the test's expectations.

Fixes flakes like [this one](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/6093943)

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/86114)
<!-- Reviewable:end -->

ajwerner committed:
sql: deflake TestUnsplit

The bug here is that we were writing the new zone config after the GC job has
already deleted the old one.

Release note: None

fqazi reviewed (approved):
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86114)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[`-- commits` line 14 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86114#-N9UcqJY-HlhN0NA2DyI):*
*nit: minor typo


<!-- Sent from Reviewable.io -->

ajwerner reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86114#-:-N9Ud7R11-pseOvnzFnL:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`-- commits` line 14 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86114#-N9UcqJY-HlhN0NA2DyI):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

*nit: minor typo
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

ajwerner commented:
bors r+
